File: day02/d2p2.hs
score :: String -> Int
score match = match[0] + match[2]
score "A X" = 3
score "A Y" = 4
score "A Z" = 8
score "B X" = 1
score "B Y" = 5
score "B Z" = 9
score "C X" = 2
score "C Y" = 6
score "C Z" = 7

read_numbers = do
    contents <- readFile "input.txt"
    -- let linesOfFile = map (\x -> read x :: String) (lines $ contents)
    let linesOfFile = (lines $ contents)
    return linesOfFile

main = do
      xs <- read_numbers
      print (sum (map score xs))

File: day02/d2p2.py
def winDraw(line):
    mapping = {"A": 0, "B": 1, "C": 2}
    mapping2 = {"X": -1, "Y": 0, "Z": 1}
    score = [1,2,3]
    return score[(mapping[line[0]]+mapping2[line[2]])%3]

count = 0
val = {"X": 0, "Y": 3, "Z": 6}
with open('input.txt', 'r') as f:
    for line in f:
        count += val[line[2]]
        count += winDraw(line)
    
    print(count)


